src/d/actor/d_a_obj_shelf.cpp

```diff
@@ -4,133 +4,288 @@
 //
 
 #include "d/actor/d_a_obj_shelf.h"
+#include "d/res/res_otana.h"
 #include "d/d_procname.h"
+#include "d/d_com_inf_game.h"
+#include "d/actor/d_a_npc_ba1.h"
+#include "m_Do/m_Do_mtx.h"
+
+#include "weak_data_1811.h" // IWYU pragma: keep
+
+Mtx daObjShelf::Act_c::M_tmp_mtx;
+
+namespace daObjShelf {
+    namespace {
+        struct Attr_c {
+            /* 0x00 */ short mVibDuration;
+            /* 0x02 */ short mVibYSpeed;
+            /* 0x04 */ float mVibYMagnitude;
+            /* 0x08 */ short mVibXSpeed;
+            /* 0x0a */ short mVibZSpeed;
+            /* 0x0c */ float mVibXMagnitude;
+            /* 0x10 */ float mVibZMagnitude;
+            /* 0x14 */ float mRotAccel;
+            /* 0x18 */ float mRotDecay;
+            /* 0x1c */ float mBounceFactor;
+            /* 0x20 */ u8 mBounceNum;
+            /* 0x21 */ u8 mRotWaitDuration;
+            /* 0x22 */ short m22;
+            /* 0x24 */ float mInitSpeed2;
+            /* 0x28 */ float mInitSpeed3;
+        };
+
+        static const Attr_c L_attr = {
+            4,
+            30000, 4.0f,
+            0x7724,0x7148,
+            200.0f,200.0f,
+            600.0f,0.02f,
+            0.5f,
+            5,0,
+            0,
+            -3500.0f,-2500.0f
+        };
+
+        inline const Attr_c & attr() { return L_attr; }
+    }
+}
+
+const char daObjShelf::Act_c::M_arcname[] = "Otana";
 
 /* 00000078-0000012C       .text CreateHeap__Q210daObjShelf5Act_cFv */
-void daObjShelf::Act_c::CreateHeap() {
-    /* Nonmatching */
+int daObjShelf::Act_c::CreateHeap() {
+    J3DModelData* model_data = (J3DModelData *)dComIfG_getObjectRes(M_arcname, OTANA_BDL_OTANA);
+    JUT_ASSERT(0x12c, model_data != 0);
+    mpModel = mDoExt_J3DModel__create(model_data, 0x80000, 0x11000022);
+    return mpModel != NULL;
 }
 
 /* 0000012C-000001A0       .text Create__Q210daObjShelf5Act_cFv */
-s32 daObjShelf::Act_c::Create() {
-    /* Nonmatching */
+BOOL daObjShelf::Act_c::Create() {
+    fopAcM_SetMtx(this, mpModel->getBaseTRMtx());
+    init_mtx();
+    fopAcM_setCullSizeBox(this, -110.0f, -55.0f, -1.0f, 110.0f, 20.0f, 110.0f);
+    mode_wait_init();
+    return TRUE;
 }
 
 /* 000001A0-0000029C       .text Mthd_Create__Q210daObjShelf5Act_cFv */
-void daObjShelf::Act_c::Mthd_Create() {
-    /* Nonmatching */
+s32 daObjShelf::Act_c::Mthd_Create() {
+    fopAcM_SetupActor(this, Act_c);
+
+    s32 phase_state = dComIfG_resLoad(&mPhs, M_arcname);
+    if (phase_state == cPhs_COMPLEATE_e) {
+        phase_state = MoveBGCreate(M_arcname, 7, dBgS_MoveBGProc_Trans, 0xb00);
```

Comment:
```suggestion
        phase_state = MoveBGCreate(M_arcname, OTANA_DZB_OTANA, dBgS_MoveBGProc_Trans, 0xb00);
```
This argument is the file index of the collision